record_android/android/src/main/kotlin/com/llfbandit/record/record/encoder/MediaCodecEncoder.kt

```diff
@@ -30,15 +32,32 @@ class MediaCodecEncoder(
         container.release()
     }
 
-    private fun createCodec(mediaFormat: MediaFormat): MediaCodec {
-        var encoder =
-            MediaCodecList(MediaCodecList.REGULAR_CODECS).findEncoderForFormat(mediaFormat)
-                ?: throw Exception("No encoder found for $mediaFormat")
+    @TargetApi(Build.VERSION_CODES.Q)
```

Comment:
For avoiding `Call requires API level 29 (current min is 21): android.media.MediaCodecInfo#getCanonicalName` warning